Fixed #35601 -- Added TelInput widget. (#18437)
django/django: docs/ref/forms/widgets.txt

    * ``template_name``: ``'django/forms/widgets/search.html'``
    * Renders as: ``<input type="search" ...>``

``TelInput``
~~~~~~~~~~~~~~

.. versionadded:: 5.2

.. class:: TelInput

    * ``input_type``: ``'tel'``
    * ``template_name``: ``'django/forms/widgets/tel.html'``
    * Renders as: ``<input type="tel" ...>``

>     The input value has no client-side validation by default because telephone
>     number formats vary so much around the world. You can add some by setting
>     ``pattern``, ``minlength``, or ``maxlength`` in the :attr:`Widget.attrs`
>     argument.

Reviewer: Small teak to make it active voice:

```suggestion
    Browsers perform no client-side validation by default because telephone
    number formats vary so much around the world. You can add some by setting
    ``pattern``, ``minlength``, or ``maxlength`` in the :attr:`Widget.attrs`
    argument.
```

    Additionally, you can add server-side validation to your form field with a
    validator like :class:`~django.core.validators.RegexValidator` or
    third-party packages, such as `django-phonenumber-field`_.

.. _django-phonenumber-field: https://django-phonenumber-field.readthedocs.io/en/latest/index.html

> 
> 

Reviewer: just one blank line
```suggestion

```

    * ``template_name``: ``'django/forms/widgets/search.html'``
    * Renders as: ``<input type="search" ...>``

``TelInput``
~~~~~~~~~~~~~~

.. versionadded:: 5.2

.. class:: TelInput

    * ``input_type``: ``'tel'``
    * ``template_name``: ``'django/forms/widgets/tel.html'``
    * Renders as: ``<input type="tel" ...>``

>     The input value is not automatically validated to a particular format
>     before the form can be submitted, because formats for telephone numbers
>     vary so much around the world.
> 
>     Client-side validations can be added by setting ``pattern``, ``minlength``
>     and ``maxlength`` keys on the :attr:`Widget.attrs` argument.

Reviewer: Suggested edits for clarity and grammatical correctness (“validations” shouldn't be plural 😉)

```suggestion
    The input value has no client-side validation by default because telephone
    number formats vary so much around the world. You can add some by setting
    ``pattern``, ``minlength``, or ``maxlength`` in the :attr:`Widget.attrs`
    argument.
```

>     Server-side validations can be achieved through the use of third-party
>     packages, such as `django-phonenumber-field`_.

Reviewer: Server-side validation isn't really related to the widget, but the field, so I’d make this an “also” section. And let’s mention the simpler options built in to Django first, probably many projects only need take phone numbers in one format.

```suggestion
    Additionally, you can add server-side validation to your form field with a
    validator like :class:`~django.core.validators.RegexValidator` or
    third-party packages, such as `django-phonenumber-field`_.
```

Author: Good call! :+1:
